contracts/0.8.9/oracle/AccountingOracle.sol

```diff
@@ -0,0 +1,760 @@
+    error NumExitedValidatorsCannotDecrease();
+    error ExitedValidatorsLimitExceeded(uint256 limitPerDay, uint256 exitedPerDay);
+    error UnsupportedExtraDataFormat(uint256 format);
+    error UnsupportedExtraDataType(uint256 dataType);
+    error MaxExtraDataItemsCountExceeded(uint256 maxItemsCount, uint256 receivedItemsCount);
+    error CannotSubmitExtraDataBeforeMainData();
+    error ExtraDataAlreadyProcessed();
+    error ExtraDataListOnlySupportsSingleTx();
+    error UnexpectedExtraDataFormat(uint256 expectedFormat, uint256 receivedFormat);
+    error UnexpectedExtraDataItemsCount(uint256 expectedCount, uint256 receivedCount);
+    error UnexpectedExtraDataIndex(uint256 expectedIndex, uint256 receivedIndex);
+    error InvalidExtraDataSortOrder();
+
+    event DataBoundariesSet(uint256 maxExitedValidatorsPerDay, uint256 maxExtraDataListItemsCount);
+    event ExtraDataSubmitted(uint256 indexed refSlot, uint256 itemsProcessed, uint256 itemsCount);
+
+    event WarnExtraDataIncompleteProcessing(
+        uint256 indexed refSlot,
+        uint256 processedItemsCount,
+        uint256 itemsCount
+    );
+
+    struct DataBoundaries {
+        uint64 maxExitedValidatorsPerDay;
+        uint64 maxExtraDataListItemsCount;
+    }
+
+    struct ExtraDataProcessingState {
+        uint64 refSlot;
+        uint16 dataFormat;
+        uint64 itemsCount;
+        uint64 itemsProcessed;
+        uint256 lastProcessedItem;
+        bytes32 dataHash;
+    }
+
+    /// @notice An ACL role granting the permission to submit the data for a committee report.
+    bytes32 public constant SUBMIT_DATA_ROLE = keccak256("SUBMIT_DATA_ROLE");
+
+    /// @notice An ACL role granting the permission to set report data safety boundaries.
+    bytes32 constant public MANAGE_DATA_BOUNDARIES_ROLE = keccak256("MANAGE_DATA_BOUNDARIES_ROLE");
+
+
+    /// @dev Storage slot: DataBoundaries dataBoundaries
+    bytes32 internal constant DATA_BOUNDARIES_POSITION =
+        keccak256("lido.AccountingOracle.dataBoundaries");
+
+    /// @dev Storage slot: ExtraDataProcessingState state
+    bytes32 internal constant EXTRA_DATA_PROCESSING_STATE_POSITION =
+        keccak256("lido.AccountingOracle.extraDataProcessingState");
+
+
+    address public immutable LIDO;
+
+    ///
+    /// Initialization & admin functions
+    ///
+
+    constructor(address lido, uint256 secondsPerSlot) BaseOracle(secondsPerSlot) {
+        if (lido == address(0)) revert LidoCannotBeZero();
+        LIDO = lido;
+    }
+
+    function initialize(
+        address admin,
+        address consensusContract,
+        uint256 consensusVersion,
+        address legacyOracle,
+        uint256 maxExitedValidatorsPerDay,
+        uint256 maxExtraDataListItemsCount
+    ) external {
+        if (admin == address(0)) revert AdminCannotBeZero();
+        _setupRole(DEFAULT_ADMIN_ROLE, admin);
+        uint256 lastProcessingRefSlot = _checkOracleMigration(legacyOracle, consensusContract);
```

Comment:
Let's allow `legacyOracle` to be zero address, because it's not needed for scratch deployment.

Reply:
done